tests/NuGet.IndexingTests/PackageAnalyzerTests.cs

```diff
@@ -0,0 +1,125 @@
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using NuGet.Indexing;
+using NuGet.IndexingTests.TestSupport;
+using Xunit;
+
+namespace NuGet.IndexingTests
+{
+    public class PackageAnalyzerTests
+    {
+        [Theory, MemberData("TheoryData")]
+        public void AddsCorrectFieldAnalyzers(string field, string text, TokenAttributes[] expected)
+        {
+            // ARRANGE
+            var analyzer = new PackageAnalyzer();
+
+            // ACT
+            var tokenStream = analyzer.TokenStream(field, new StringReader(text));
+            var tokenAttributes = tokenStream.GetTokenAttributes().ToArray();
+
+            // ASSERT
+            Assert.Equal(expected, tokenAttributes);
+        }
+
+        public static IEnumerable<object[]> TheoryData
+        {
+            get
+            {
+                yield return new object[]
+                {
+                    "Id",
+                    "AaBbCc",
+                    new[]
+                    {
+                        new TokenAttributes("aabbcc", 0, 6)
+                    }
+                };
+
+                yield return new object[]
+                {
+                    "IdAutocomplete",
+                    "AaB",
+                    new[]
+                    {
+                        new TokenAttributes("a", 0, 1, 1),
+                        new TokenAttributes("aa", 0, 2, 1),
+                        new TokenAttributes("aab", 0, 3, 1),
+                        new TokenAttributes("a", 0, 1, 1),
+                        new TokenAttributes("aa", 0, 2, 1),
+                        new TokenAttributes("b", 2, 3, 1)
+                    }
+                };
+
+                yield return new object[]
+                {
+                    "TokenizedId",
+                    "AaBb",
+                    new[]
+                    {
+                        new TokenAttributes("aabb", 0, 4, 1),
+                        new TokenAttributes("aa", 0, 2, 0),
+                        new TokenAttributes("bb", 2, 4, 1)
+                    }
+                };
+
+                yield return new object[]
+                {
+                    "Version",
+                    "01.002.0003",
+                    new[]
+                    {
+                        new TokenAttributes("1.2.3", 0, 11)
+                    }
+                };
+
+                yield return GetDescriptionTestCase("Title");
+                yield return GetDescriptionTestCase("Description");
+                yield return GetDescriptionTestCase("Summary");
+                yield return GetDescriptionTestCase("Authors");
+                
+                yield return new object[]
+                {
+                    "Owner",
+                    "AAA Bbb",
+                    new[]
+                    {
+                        new TokenAttributes("aaa bbb", 0, 7)
```

Comment:
I don't think so. All four of these fields have similar types of content and should be tokenized similarly (DotTokenizer, case insensitive, remove stop words, split on camel case). However, you have a lot more context here so please correct me.